pkg/virt-handler/vm.go

```diff
@@ -593,6 +596,80 @@ func mapPVToDisk(disk *v1.Disk, pv *k8sv1.PersistentVolume) (*v1.Disk, error) {
 	}
 }
 
+func (d *VirtualMachineController) getUnusedSerialPort(vm *v1.VirtualMachine) (uint, error) {
+	// look for an unclaimed serial port.
+	for i := 0; i < 100; i++ {
+		claimed := false
+		for _, serial := range vm.Spec.Domain.Devices.Serials {
+			if serial.Target == nil || serial.Target.Port == nil {
+				continue
+			}
+
+			if *serial.Target.Port == uint(i) {
+				claimed = true
+				break
+			}
+		}
+
+		if claimed == false {
+			return uint(i), nil
+		}
+	}
+
+	return 0, goerror.New("No serial port found for console")
+}
+
+func (d *VirtualMachineController) cleanupConsoleSockets(vm *v1.VirtualMachine) error {
+	// TODO this can go away once qemu is in the pods mount namespace.
+	namespace := vm.ObjectMeta.Namespace
+	name := vm.ObjectMeta.Name
+	unixPath := fmt.Sprintf("%s-private/%s/%s", d.virtShareDir, namespace, name)
+	return diskutils.RemoveFile(unixPath)
+}
+
+func (d *VirtualMachineController) MapConsoleAccess(vm *v1.VirtualMachine) *v1.VirtualMachine {
+	for _, console := range vm.Spec.Domain.Devices.Consoles {
+		if console.Type == "pty" && console.Target == nil {
```

Comment:
@rmohr how will you allow to expose serial consoles in the api redesign?

Reply:
Right now, console will not be part of the api on the redesign. The pty will be hardcoded until we have further requirements/stories (same for spice/video).